### ISSUE TITLE
SSL Error with self-signed certificate in dev localhost when using Chrome 73

### ISSUE THREAD

**Issue by happymango7** [puma/puma#1756]
### Steps to reproduce

1) Localhost dev environment on macOS 10.14.2

2) Setup and configure a self-signed certificate

3) Try to connect with Chrome 73

### Expected behavior

Should connect to https://localhost:3000 in Chrome 73

### Actual behavior

Error on rails console:
```
2019-03-28 07:44:11 -0700: SSL error, peer: ::1, peer cert: , #<Puma::MiniSSL::SSLError: OpenSSL error: error:141F7065:SSL routines:final_key_share:no suitable key share - 337604709>
```

Error in Chrome  73 browser
```
This site can’t be reached localhost is currently unreachable.
...
ERR_SSL_VERSION_INTERFERENCE
```

### System configuration

**Ruby version**:
2.4.4
**Rails version**:
5.1.5
**Puma version**:
3.12.1

**Comment by happymango7** [comment]
@schneems 
Just an FYI, this commit seems working and it was merged into master branch, but I wonder why this was not released in 3.12.1.
https://github.com/puma/puma/commit/fc97fdaab2290f93e348a23b5bec9b01e1f195bc

**Comment by happymango7** [comment]
`master` branch resolved this issue. But when will `master` branch get released?

**Comment by siddhant-mohan** [comment]
when is this going to be released?

**Comment by siddhant-mohan** [comment]
when is this going to be released?

**Comment by truongnmt** [comment]
I'm on Chrome 79, puma 3.12.2 and I still have this issue

**Comment by nateberkopec** [comment]
Fixed in 4.0.
